### DISCUSSION TITLE
Is it possible to replace specific text pattern ignoring AST?

### DISCUSSION THREAD

**Discussion by hitochan777** [ast-grep/ast-grep#1010] [Q&A] [answered]
I would like to replace `vuetify.breakpoint` with `vuetify.display` but I don't care about in what context it appears.
I don't have to use `ast-grep` for this purpose and instead use something like `sed`.
But there are many other replacements that requires `ast-grep` and I want to use a single tool if possible.

I thought it can use `regex` for this purpose but can't make it work.
[Reproduction](https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6Imh0bWwiLCJxdWVyeSI6IiIsInJld3JpdGUiOiIkQT8uKCkiLCJjb25maWciOiJpZDogZm9vXG5ydWxlOlxuICByZWdleDogXCIkdnVldGlmeS5icmVha3BvaW50XCJcbmZpeDogXCIkdnV0aWZ5LmRpc3BsYXlcIiIsInNvdXJjZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdiB2LWlmPVwiJHZ1ZXRpZnkuYnJlYWtwb2ludC54c1wiPnhzPC9kaXY+XG4gIDxkaXYgdi1lbHNlPm5vdCB4czwvZGl2PlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdD5cbmV4cG9ydCBkZWZhdWx0IHtcbiAgbWV0aG9kczoge1xuICAgIGZvbygpIHtcbiAgICAgICAgZG9Tb21ldGhpbmcodGhpcy4kdnVldGlmeS5icmVha3BvaW50KTtcbiAgICB9XG59XG48L3NjcmlwdD4ifQ==)

Thanks in advance!

**Comment by HerringtonDarkholme** [comment]
You rule does not match because `$` is a special character in regular expression. You need to escape it `\\$`

https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6Imh0bWwiLCJxdWVyeSI6IiIsInJld3JpdGUiOiIkQT8uKCkiLCJjb25maWciOiJpZDogZm9vXG5ydWxlOlxuICBraW5kOiBhdHRyaWJ1dGVfdmFsdWVcbiAgcmVnZXg6IFwiXFxcXCR2dWV0aWZ5LmJyZWFrcG9pbnRcIlxuZml4OiBcIiR2dXRpZnkuZGlzcGxheVwiIiwic291cmNlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IHYtaWY9XCIkdnVldGlmeS5icmVha3BvaW50LnhzXCI+eHM8L2Rpdj5cbiAgPGRpdiB2LWVsc2U+bm90IHhzPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0PlxuZXhwb3J0IGRlZmF1bHQge1xuICBtZXRob2RzOiB7XG4gICAgZm9vKCkge1xuICAgICAgICBkb1NvbWV0aGluZyh0aGlzLiR2dWV0aWZ5LmJyZWFrcG9pbnQpO1xuICAgIH1cbn1cbjwvc2NyaXB0PiJ9

**Reply by hitochan777** [reply]
That was a silly mistake. Thanks for the quick response as always!

**Reply by hitochan777** [reply]
But I also want to match the text inside `<script>` tag.
That is why I did not specify `kind` tag.
Is it possible with one rule?

**Reply by HerringtonDarkholme** [reply]
Yes, it can! But it is somewhat ugly because you need to add all possible kinds in the `any` rule.

https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6Imh0bWwiLCJxdWVyeSI6IiIsInJld3JpdGUiOiIkQT8uKCkiLCJjb25maWciOiJpZDogZm9vXG5ydWxlOlxuICBhbnk6XG4gIC0ga2luZDogYXR0cmlidXRlX3ZhbHVlXG4gIC0ga2luZDogcmF3X3RleHRcbiAgcmVnZXg6IFwiXFxcXCR2dWV0aWZ5LmJyZWFrcG9pbnRcIlxuZml4OiBcIiR2dXRpZnkuZGlzcGxheVwiIiwic291cmNlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IHYtaWY9XCIkdnVldGlmeS5icmVha3BvaW50LnhzXCI+eHM8L2Rpdj5cbiAgPGRpdiB2LWVsc2U+bm90IHhzPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0PlxuZXhwb3J0IGRlZmF1bHQge1xuICBtZXRob2RzOiB7XG4gICAgZm9vKCkge1xuICAgICAgICBkb1NvbWV0aGluZyh0aGlzLiR2dWV0aWZ5LmJyZWFrcG9pbnQpO1xuICAgIH1cbn1cbjwvc2NyaXB0PiJ9

**Reply by hitochan777** [reply]
Thanks!

Can I keep surrounding text after the replacement? 
Currently `$vuetify.breakpoint.xs` is replaced with `vuetify.display`.
But I expect it to be replaced with `$vuetify.display.xs`.

**Reply by HerringtonDarkholme** [reply, marked as answer]
You can try using regex + replace transformation https://ast-grep.github.io/reference/yaml/transformation.html#replace

**Reply by hitochan777** [reply]
I was finally able to make it work!
Thanks for your patience!

https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6Imh0bWwiLCJxdWVyeSI6IiIsInJld3JpdGUiOiIkQT8uKCkiLCJjb25maWciOiJpZDogZm9vXG5ydWxlOlxuICBhbnk6XG4gIC0ga2luZDogZnJhZ21lbnRcbiAgcmVnZXg6IFwiXFxcXCR2dWV0aWZ5LmJyZWFrcG9pbnRcIlxuICBwYXR0ZXJuOiBcIiRBXCJcbnRyYW5zZm9ybTpcbiAgQjpcbiAgICByZXBsYWNlOlxuICAgICAgYnk6IFwiJCR2dWV0aWZ5LmRpc3BsYXlcIlxuICAgICAgcmVwbGFjZTogXCJcXFxcJHZ1ZXRpZnkuYnJlYWtwb2ludFwiXG4gICAgICBzb3VyY2U6ICRBXG5maXg6XG4gICRCIiwic291cmNlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IHYtaWY9XCIkdnVldGlmeS5icmVha3BvaW50LnhzXCI+eHM8L2Rpdj5cbiAgPGRpdiB2LWVsc2U+bm90IHhzPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0PlxuZXhwb3J0IGRlZmF1bHQge1xuICBtZXRob2RzOiB7XG4gICAgZm9vKCkge1xuICAgICAgICBkb1NvbWV0aGluZyh0aGlzLiR2dWV0aWZ5LmJyZWFrcG9pbnQpO1xuICAgIH1cbn1cbjwvc2NyaXB0PiJ9
